pandas/tests/test_index.py

```diff
@@ -2680,6 +2688,14 @@ def test_get_indexer(self):
         with self.assertRaisesRegexp(ValueError, 'different freq'):
             idx.asfreq('D').get_indexer(idx)
 
+    def test_repeat(self):
+        # GH10183
+        idx = pd.period_range('2000-01-01', periods=3, freq='D')
+        res = idx.repeat(3)
```

Comment:
problem here is that I explicitly wanted to test the `freqstr` as well (as this is what was actually not working) 
(but I can have both, and put a generic one in Base as well)